src/main/java/org/terasology/additionalRails/action/HooverCartAction.java

```diff
@@ -55,7 +55,7 @@ public void update(float delta) {
                     ItemComponent itemComponent = fuelSlot.getComponent(ItemComponent.class);
                     BlockItemComponent biComponent = fuelSlot.getComponent(BlockItemComponent.class);
                     //add 5 energy units for 1 torch
-                    if (biComponent.blockFamily.getURI().toString().equalsIgnoreCase("core:torch")) {
+                    if (biComponent.blockFamily.getURI().toString().toLowerCase().endsWith("torch")) {
```

Comment:
Could you please add a helper method `isTorch(BlockItemComponent blockItem)` containing this check. I hope that will make it easier to find the check later on in case we find a better way to check for torches...
```suggestion
                    if (isTorch(biComponent)) {
```

Reply:
```java
private boolean isTorch(BlockItemComponent blockItem) {
    return blockItem.blockFamily.getURI().toString().toLowerCase().endsWith("torch");
}
```